### ISSUE TITLE
`vec2<u32>` not interpolated flat

### ISSUE THREAD

**Issue by N4tus** [gfx-rs/wgpu#1751]
**Description**
A User-defined IO of a vector integer type should always be interpolated flat. However if I use a `[[location(0)]] data: vec2<u32>` the value the fragment shader receives is always the null-vector.  

**Repro steps**  
This is the shader I use: 
```
[[block]] 
struct Uniforms {
    view_proj: mat4x4<f32>;
};
[[group(0), binding(0)]] 
var<uniform> uniforms: Uniforms;

struct VertexInput {
    [[location(0)]] entity: vec2<u32>;
    [[location(1)]] position: vec2<f32>;
};

struct VertexOutput {
    [[builtin(position)]] clip_position: vec4<f32>;
    [[location(0)]] entity: vec2<u32>;
};

[[stage(vertex)]]
fn main(
    model: VertexInput,
) -> VertexOutput {
    var out: VertexOutput;
    out.clip_position = uniforms.view_proj * vec4<f32>(model.position, 0.0, 1.0);
    // out.entity = vec2<f32>(f32(model.entity.x), f32(model.entity.y));
    out.entity = model.entity;
    return out;
}

[[stage(fragment)]]
fn main(in: VertexOutput) -> [[location(0)]] vec2<u32> {
    // return vec2<u32>(u32(in.entity.x), u32(in.entity.y));
    return in.entity;
}
```
`entity` is a `vec2<u32>` and I expect it to be interpolated flat. My current workaround is to convert it into a `vec2<f32>` and back and hope the precision of a float is good enough.

**Platform**
Running on Ubuntu 21.04, wgpu 0.8 and 0.9

**Comment by kvark** [comment]
> My current workaround is to convert it into a vec2<f32> and back and hope the precision of a float is good enough.

You should be able to specify the "flat" interpolation as `interpolate(flat)` explicitly until we fix this.

For our reference, the relevant bit of the spec is:
> User-defined IO of scalar or vector integer type is always [[interpolate(flat)]] and, therefore, must not be specified in a WGSL program.

**Comment by cwfitzgerald** [comment]
This should be fixed, enough stuff has happened since 0.9
